client/QuickStart/Program.cs

```diff
@@ -0,0 +1,767 @@
+                            $"File '{configFilePath}' is missing required fields: {string.Join(", ", missing.Select(item => item.name))}");
+
+                    if (config.Data is null)
+                        ErrorHandler($"File '{configFilePath}' is missing required fields: Data");
+                    return config;
+                }
+            }
+
+            catch (Exception ex)
+            {
+                ErrorHandler($"Couldn't read config file: '{configFilePath}'", ex);
+            }
+
+            return null;
+        }
+
+
+        /// <summary>
+        /// Generates Kusto Connection String based on given Authentication Mode.
+        /// </summary>
+        /// <param name="clusterUrl"> Cluster to connect to.</param>
+        /// <param name="authenticationMode">User Authentication Mode, 
+        ///                                  Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)</param>
+        /// <param name="certificatePath">Given certificate path</param>
+        /// <param name="certificatePassword">Given certificate password</param>
+        /// <param name="applicationId">Given application id</param>
+        /// <param name="tenantId">Given tenant id</param>
+        /// <returns>A connection string to be used when creating a Client</returns>
+        private static KustoConnectionStringBuilder GenerateConnectionString(string clusterUrl,
+            string authenticationMode, string certificatePath, string certificatePassword, string applicationId,
+            string tenantId)
+        {
+            //Learn More: For additional information on how to authorize users and apps in Kusto see:
+            //https://docs.microsoft.com/azure/data-explorer/manage-database-permissions
+            switch (authenticationMode)
+            {
+                case "UserPrompt":
+                    // Prompt user for credentials
+                    return new KustoConnectionStringBuilder(clusterUrl).WithAadUserPromptAuthentication();
+
+                case "ManagedIdentity":
+                    // Authenticate using a System-Assigned managed identity provided to an azure service, or using a
+                    // User-Assigned managed identity. For more information, see
+                    // https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
+                    return CreateManagedIdentityConnectionString(clusterUrl);
+
+                case "AppKey":
+                    // Learn More: For information about how to procure an AAD Application, see:
+                    // https://docs.microsoft.com/azure/data-explorer/provision-azure-ad-app
+                    // TODO (config - optional): App ID & tenant, and App Key to authenticate with
+                    return new KustoConnectionStringBuilder(clusterUrl).WithAadApplicationKeyAuthentication(
+                        Environment.GetEnvironmentVariable("APP_ID"),
```

Comment:
Some inconsistency here - why are these env vars, but for the certificate you use config?

Reply:
This was based on the python implementation, while the certificate was based on our teams discussion. checking on the python  implementation, it seems we used env vars in the certificate one as well. which way should i alter the functions then? is it something that will come with the config file, or is it a part of the env?

Reply:
The general rules are that credentials is something you want to store in the env, and not in a config file (for example you may want to commit your config file, but you don't want to commit your credentials), so the certificate should be in env

Reply:
Ok, will change to env based. However, it seems like in C# we have no different functions for public and private certs as in Python, an no actual Parmeter to receive them. should i use a different authenticate method? (relates back to Yochai's point on this)